File: akuntansi_db/app/Models/SalesHeader.php
<?php

namespace App\Models;

use Carbon\Carbon;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class SalesHeader extends Model
{
    use HasFactory;

    protected $fillable = [
        'transaction_date',
        'c_id',
        'gross_amount',
        'net_amount',
        'note',
    ];

    public function setTransactionDateAttribute($value)
    {
        $this->attributes['transaction_date'] = Carbon::parse($value);
    }
}

File: akuntansi_flut/lib/services/model/sales_header.dart
class SalesHeader {
  int? id;
  String? transactionDate;
  int? cId;
  int? grossAmount;
  int? netAmount;
  String? note;
  String? createdAt;
  String? updatedAt;
  String? cName;

  SalesHeader({this.id, this.transactionDate, this.cId, this.grossAmount, this.netAmount, this.note, this.createdAt, this.updatedAt, this.cName});

  SalesHeader.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    transactionDate = json['transaction_date'];
    cId = json['c_id'];
    grossAmount = json['gross_amount'];
    netAmount = json['net_amount'];
    note = json['note'];
    createdAt = json['created_at'];
    updatedAt = json['updated_at'];
    cName = json['c_name'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['transaction_date'] = transactionDate;
    data['c_id'] = cId;
    data['gross_amount'] = grossAmount;
    data['net_amount'] = netAmount;
    data['note'] = note;
    data['created_at'] = createdAt;
    data['updated_at'] = updatedAt;
    data['c_name'] = cName;
    return data;
  }
}
